src/recipe/Baked.Recipe.Service.Application/DataAccess/Sqlite/SQLiteConfiguration.cs

```diff
@@ -19,7 +19,7 @@ public SQLiteConfiguration InMemory()
     {
         Raw("connection.release_mode", "on_close");
 
-        return ConnectionString(c => c.Is(new SqliteConnectionStringBuilder { DataSource = ":memory:" }.ToString()));
+        return ConnectionString(c => c.Is(new SqliteConnectionStringBuilder { DataSource = ":memory:", Mode = SqliteOpenMode.Memory, Cache = SqliteCacheMode.Shared }.ToString()));
```

Comment:
why?

Reply:
after the changes made to `DataAccess` layer, we needed database to persist and be shared between multiple connections

Reply:
great! i didn't know this was possible. now that we have this option, let's not share a session and connection throughout the whole test run. instead of this, let's create a session as well as a transaction on set up, and dispose the scope on tear down.